File: backend/services/product_service.go
package services

import (
	"kasirgo-backend/models"
	"kasirgo-backend/repositories"
)

type ProductService struct {
	productRepo  *repositories.ProductRepository
	categoryRepo *repositories.CategoryRepository
}

func NewProductService() *ProductService {
	return &ProductService{
		productRepo:  repositories.NewProductRepository(),
		categoryRepo: repositories.NewCategoryRepository(),
	}
}

func (s *ProductService) CreateProduct(product *models.Product) error {
	return s.productRepo.Create(product)
}

func (s *ProductService) GetProductByID(id string) (*models.Product, error) {
	return s.productRepo.FindByID(id)
}

func (s *ProductService) GetProductsBySKU(sku string) (*models.Product, error) {
	return s.productRepo.FindBySKU(sku)
}

func (s *ProductService) GetProductsByStore(storeID string, activeOnly bool) ([]models.Product, error) {
	return s.productRepo.FindByStoreID(storeID, activeOnly)
}

func (s *ProductService) GetProductsByCategory(categoryID string) ([]models.Product, error) {
	return s.productRepo.FindByCategoryID(categoryID)
}

func (s *ProductService) UpdateProduct(product *models.Product) error {
	return s.productRepo.Update(product)
}

func (s *ProductService) DeleteProduct(id string) error {
	return s.productRepo.Delete(id)
}

File: kasirgo/lib/app/data/services/product_service.dart
import 'package:get/get.dart';
import '../models/product_model.dart';
import 'api_service.dart';

class ProductService extends ApiService {
  Future<Response> getProducts({bool activeOnly = true}) async {
    return await get('/products?active=$activeOnly');
  }

  Future<Response> getProductBySKU(String sku) async {
    return await get('/products/sku?sku=$sku');
  }

  Future<Response> getProductsByCategory(String categoryId) async {
    return await get('/categories/$categoryId/products');
  }

  Future<Response> createProduct(ProductModel product) async {
    return await post('/products', product.toJson());
  }

  Future<Response> updateProduct(String id, ProductModel product) async {
    return await put('/products/$id', product.toJson());
  }

  Future<Response> deleteProduct(String id) async {
    return await delete('/products/$id');
  }
}
